frame/stake-tracker/src/lib.rs

```diff
@@ -0,0 +1,208 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2023 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Stake Tracker Pallet
+//!
+//! The Stake Tracker pallet is used to maintain a sorted list of voters by listening to the events
+//! that Staking emits.
+//!
+//! - [`Config`]
+//! - [`Pallet`]
+//!
+//! ## Overview
+//!
+//! The goal of Stake Tracker is to reduce the burden of Staking pallet by taking care of the
```

Comment:
I am not going to comment on this, I think I have already annoyed you enough :D I think we need to have a session on how to write docs and make sure we are aligned :) but, perhaps with my existing feedback, you can guess what wrong "assumptions" are being made here too.

Reply:
I understand your point, however it still feels wrong to omit this detail as this is exactly the purpose of this pallet - taking over some functionality that used to live in Staking. It can be used for nothing else as it's pretty tightly coupled with Staking logic and assumptions about the SortedListProvider(s) contents it's using.

At the very least this is now documented in here.

Reply:
A good middle-ground is this: 

A good document should _start_ at its _most abstract form_, and move toward being _concrete_. 

In this case, you should first talk about what a pallet, or an interface is doing without making any assumption (what I am asserting that you are missing right now).

Then, you can gradually move toward a "specific usage" of this pallet that is in conjuncture with the staking pallet, etcetera.